Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationEmail.cshtml.cs

```diff
@@ -19,9 +21,19 @@ public class OrganisationEmailModel(ISession session) : RegistrationStepModel(se
     [BindProperty]
     public bool? RedirectToSummary { get; set; }
 
-    public void OnGet()
+    public async Task OnGet()
     {
         EmailAddress = RegistrationDetails.OrganisationEmailAddress;
+
+        if ((RegistrationDetails.OrganisationScheme == OrganisationSchemeType.CharityCommissionEnglandWales) && (string.IsNullOrEmpty(EmailAddress)))
+        {
+            var details = await charityCommissionApi.GetCharityDetails(RegistrationDetails.OrganisationIdentificationNumber!);
```

Comment:
Rather than causing a potential runtime exception with `!`, can we move the check to the condition above to ensure `RegistrationDetails.OrganisationIdentificationNumber != null` please?